### PR TITLE
Errors on tracker popups

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Adds a dumb component for error in tracker and hooks it up to where we've been putting errors in the store.

Needs testing, let me know if you can think of a good way of testing this. The related ticket didn't have anything in the desktop log.

### CODE REVIEW

**Comment by chrisnojima** [comment]
are you going to do the timeout on getting a finish call on this pr or a different one @MarcoPolo

**Comment by MarcoPolo** [comment]
I'll do it on this one

**Comment by MarcoPolo** [comment]
Okay this is ready for review

File: desktop/renderer/remote-tracker.js [hunk 1]
```diff
@@ -47,6 +48,7 @@ class RemoteTracker extends Component<void, Props, void> {
             component='tracker'
             username={username}
             startTimer={this.props.startTimer}
+            errorRetry={() => this.props.errorRetry(username)}
```
Comment (MarcoPolo) [inline desktop/renderer/remote-tracker.js]: So hacky... 
We need to do this to make sure the side effects run in the main window.

With sagas all action creators are pure and we won't need this.

File: shared/tracker/error.desktop.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        trackerState={'error'}
+        currentlyFollowing={false}
+        headerStyle={{position: undefined}}
+        loggedIn={false}
+      />
+      <Text style={errorTextStyle} type='Error'>
+        {errorMessage}
+      </Text>
+      <Box style={retryStyle}>
+        <Button onClick={() => onRetry()} type='Primary' label='Retry fetching identity' />
```
Comment (MarcoPolo) [inline shared/tracker/error.desktop.js]: Without this and a proxied function, remote windows hang. No clue why.

File: shared/actions/tracker.js [hunk 3]
```diff
@@ -171,9 +171,20 @@ export function registerIdentifyUi (): TrackerActionCreator {
...
+
+      const onStart = (username) => {
+        trackerTimeoutError = setTimeout(() => {
+          dispatch({type: Constants.identifyFinished, error: true, payload: {username, error: 'Identify timed out'}})
+        }, trackerTimeout)
+      }
+
       const session: Session = engine().createSession(
-        serverCallMap(dispatch, getState, false, () => {
+        serverCallMap(dispatch, getState, false, onStart, () => {
```
Comment (chromakode) [inline shared/actions/tracker.js]: :sheep: Maybe move this arrow function out into an `onFinish` const to match `onStart`

File: shared/tracker/error.desktop.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import {Box, Button, Text} from '../common-adapters'
+import {globalStyles, globalMargins} from '../styles'
+import type {Props} from './error'
+
+function TrackerError ({errorMessage, onRetry, onClose}: Props) {
+  return (
+    <Box style={{...globalStyles.flexBoxColumn, flex: 1}}>
+      <Header
+        reason={"Error in getting identity information :'("}
```
Comment (chromakode) [inline shared/tracker/error.desktop.js]: :sheep: No need for curly braces here?

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits c5bc8e610548...743e9c94a68e introduced some visual changes on linux:
 * Added: **Tracker-Tracker_Error**  
   Tracker-Tracker_Error rendered

**Pull request merged**